pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -111,9 +109,16 @@ func (g *Gatherer) GetName() string {
 // GetGatheringFunctions returns gathering functions that should be run considering the conditions
 // + the gathering function producing metadata for the conditional gatherer
 func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gatherers.GatheringClosure, error) {
+	// later the config will be downloaded from an external source
+	errs := validateGatheringRules(gatheringRules)
+	if len(errs) > 0 {
+		klog.Error("got invalid config for conditional gatherer: %v", utils.SumErrors(errs))
```

Comment:
It should be `klog.Errorf`. Anyway, I got the following message when I tried to introduce some validation failure (and then noticed it in the tests...lol):
```
conditional_gatherer.go:116] got invalid config for conditional gatherer: Key: 'GatheringRule.Conditions[].Type' Error:Field validation for 'Conditions[].Type' failed on the 'is_valid' tag
```
I don't find this message very helpful TBH, but this might change in the future. Maybe the `GatheringRule` will have a name....

Reply:
Well, the underlying error has a lot of information, but it's not exposed by the default `Error()` function, I'll try to find a way to make it more comprehensible

Reply:
@tremes I think now (I turned them to JSONs) is the most we can get, we could also add the value itself to the error, but it maybe not the best idea, not sure.